packages/itwin/grouping-mapping-widget/src/widget/utils.ts

```diff
@@ -2,14 +2,60 @@
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
+import type { ReactElement } from "react";
+
 export type CreateTypeFromInterface<Interface> = {
   [Property in keyof Interface]: Interface[Property];
 };
 
-export function debounce<F extends (...args: any[]) => void>(f: F, delay: number) {
+export function debounce<F extends (...args: any[]) => void>(
+  f: F,
+  delay: number,
+) {
   let timer: number | undefined;
   return (...args: any[]) => {
     timer && window.clearTimeout(timer);
     timer = window.setTimeout(f, delay, ...args);
   };
 }
+
+/**
+ * Custom UI provider type definition
+ */
+export interface CustomUIProvider {
```

Comment:
It's weird to see that such important interfaces live in the `utils.ts` file. I think they deserve their own file. `CustomUIProvider.ts`?